src/Jackalope/Node.php

```diff
@@ -773,9 +785,10 @@ public function getPropertiesValues($nameFilter = null, $dereference = true)
         $this->checkState();
 
         // OPTIMIZE: do not create properties in constructor, go over array here
-        $names = self::filterNames($nameFilter, array_keys($this->properties));
+        $names = self::filterNames($nameFilter, array_keys($this->propertyValues));
         $result = array();
         foreach ($names as $name) {
+            $this->loadProperty($name);
```

Comment:
How cheap is it to get the type of a property? At this place there's only the name, so I have to load something either way. I am not sure if this causes a notable performance gain, and therefore is worth the duplication of this logic.